Dockerfile

```diff
@@ -1,49 +1,36 @@
-FROM node:16-slim
+#
+# Build
+#
+FROM node:16-slim AS builder
 
-# Install node_modules into a different directory to avoid npm/npm#9863.
-RUN mkdir -p /srv/node
-COPY package.json /srv/node/
-COPY yarn.lock /srv/node/
 WORKDIR /srv/node
+COPY package.json yarn.lock /srv/node/
 
-# This file has been downloaded from: https://dl.yarnpkg.com/debian/pubkey.gpg
-COPY docker/etc/pki/yarnpkg.gpg.key /etc/pki/yarnpkg.gpg.key
-
-RUN buildDeps=' \
-    git \
-    yarn \
-    python \
-    build-essential \
-    ' && \
-    # `apt-transport-https` is required to use https deb repositories
-    apt-get update -y && \
-    apt-get install -y --no-install-recommends gnupg ca-certificates apt-transport-https && \
-    # configure Yarn repository, see: https://yarnpkg.com/en/docs/install#linux-tab
-    apt-key add /etc/pki/yarnpkg.gpg.key && \
-    echo "deb https://dl.yarnpkg.com/debian/ stable main" > /etc/apt/sources.list.d/yarn.list && \
-    # the base image installs yarn, let's be sure we use ours
-    rm -f /usr/local/bin/yarn /usr/local/bin/yarnpkg && \
-    # install deps
-    apt-get update -y && \
-    apt-get install -y --no-install-recommends $buildDeps && \
-    yarn install --pure-lockfile && \
-    # cleanup
-    yarn cache clean && \
-    # We need to keep yarn to run the container
-    apt-get purge -y git python build-essential && \
-    apt autoremove -y && \
-    rm -rf /var/lib/apt/lists/*
```

Comment:
Removing the extra `yarn` steps is ok, but the builder needs at least `python`, possibly more. Otherwise we fail with:
```
Command: node-gyp rebuild
Arguments: 
Directory: /srv/node/node_modules/unix-dgram
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@9.1.0
gyp info using node@16.18.1 | linux | x64
gyp ERR! find Python 
gyp ERR! find Python Python is not set from command line or npm configuration
gyp ERR! find Python Python is not set from environment variable PYTHON
gyp ERR! find Python checking if \"python3\" can be used
gyp ERR! find Python - \"python3\" is not in PATH or produced an error
gyp ERR! find Python checking if \"python\" can be used
gyp ERR! find Python - \"python\" is not in PATH or produced an error
gyp ERR! find Python 
gyp ERR! find Python **********************************************************
gyp ERR! find Python You need to install the latest version of Python.
gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
gyp ERR! find Python you can try one of the following options:
gyp ERR! find Python - Use the switch --python=\"/path/to/pythonexecutable\"
gyp ERR! find Python   (accepted by both node-gyp and npm)
gyp ERR! find Python - Set the environment variable PYTHON
gyp ERR! find Python - Set the npm configuration variable python:
gyp ERR! find Python   npm config set python \"/path/to/pythonexecutable\"
gyp ERR! find Python For more information consult the documentation at:
gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
gyp ERR! find Python **********************************************************
gyp ERR! find Python 
gyp ERR! configure error 
gyp ERR! stack Error: Could not find any Python installation to use
gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:330:47)
gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:159:21)
gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:202:16)
gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:294:16)
gyp ERR! stack     at exithandler (node:child_process:410:5)
gyp ERR! stack     at ChildProcess.errorhandler (node:child_process:422:5)
gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
gyp ERR! stack     at onErrorNT (node:internal/child_process:485:16)
gyp ERR! stack     at processTicksAndRejections (node:internal/process/task_queues:83:21)
gyp ERR! System Linux 5.15.0-52-generic
gyp ERR! command \"/usr/local/bin/node\" \"/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js\" \"rebuild\"
gyp ERR! cwd /srv/node/node_modules/unix-dgram
gyp ERR! node -v v16.18.1
gyp ERR! node-gyp -v v9.1.0
gyp ERR! not ok"
```

Reply:
I see the same error in the latest successful CircleCI build for the master branch: https://app.circleci.com/pipelines/github/mozilla/addons-frontend/8323/workflows/a76cabe1-2337-4426-ac46-3c8b0f44e6f8/jobs/59027?invite=true&invite=true#step-108-658

Even in the latest release build: https://app.circleci.com/pipelines/github/mozilla/addons-frontend/8314/workflows/4577c0db-7785-4478-af6c-a2878f9ec06d/jobs/58960?invite=true#step-108-658

This error must have been there ever since `node-gyp.js` stopped supporting Python v2 because the current Dockerfile will only install Python v2.

This makes me wonder if the application doesn't need the `unix-dgram` module, should we remove it from the dependency list?

We could install Python v3 and the build error may go away, but that just means introducing more bits that the app doesn't need.

Reply:
It's a transient dependency, not directly included by us (I'll retest once we have code-manager working)